### PR TITLE
Memory Pair Game

### PR DESCRIPTION
**Pull request by vakulinina**
### Memory Pair Game

[Demo](https://vakulinina.github.io/kottans-memory-pair-game/)
[Code base](https://github.com/vakulinina/kottans-memory-pair-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by OleksiyRudenko** [comment]
@vakulinina nicely done.

Spotted a bug. If the third card clicked in row while two first didn't flip back yet, one card remains flipped just like it's been selected. However it never gets compared to any other card.

image

More over any following matching pairs remain face top.

image

**Comment by vakulinina** [comment]
@OleksiyRudenko styles on your screenshot look a bit weird, could you please say what browser it is, so I could check?

**Comment by OleksiyRudenko** [comment]
@vakulinina Version 87.0.4280.88 (Official Build) (64-bit)

File: submissions/vakulinina/memory-pair-game/main.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+];
+
+let firstFlip;
+let secondFlip;
+let pairsOpen = 0;
+
+const cards = cardIds.flatMap(id => [createCard(id), createCard(id)]);
+
+function createCard(id) {
+  let newCard = document.createElement('li');
```
Comment (babayK0) [inline submissions/vakulinina/memory-pair-game/main.js:22]: why `let` ?

File: submissions/vakulinina/memory-pair-game/main.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  newCard.classList.add('card');
+  newCard.id = id;
+  newCard.innerHTML = `
+    <div class="card-front">
+      <img src="img/icon-cat.svg" alt="">
+    </div>
+    <div class="card-back">
+      <img src="img/${id}.png" alt="">
```
Comment (babayK0) [inline submissions/vakulinina/memory-pair-game/main.js:30]: try to store full src in your obj/array (what if one image must be in .jpg format and another in .png ?)

File: submissions/vakulinina/memory-pair-game/main.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+function restartGame() {
+  pairsOpen = 0;
+  cards.forEach(card => card.classList.remove('flip'));
+  renderCards();
+}
+
+function shuffleCards(cards) {
+  for (var i = cards.length - 1; i > 0; i--) {
+    var j = Math.floor(Math.random() * (i + 1));
```
Comment (babayK0) [inline submissions/vakulinina/memory-pair-game/main.js:80]: `var` ? =)

File: submissions/vakulinina/memory-pair-game/main.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  if (firstFlip.id !== secondFlip.id) {
+    setTimeout(function () {
+      firstFlip.classList.remove('flip');
+      secondFlip.classList.remove('flip');
+      firstFlip = null;
+      secondFlip = null;
+    }, 600)
+  } else {
+    firstFlip = null;
+    secondFlip = null;
```
Comment (babayK0) [inline submissions/vakulinina/memory-pair-game/main.js:58]: looks like you can move out this block  from if/else
```
firstFlip = null;
secondFlip = null;
```

**Review by babayK0 — CHANGES_REQUESTED** [review]
Good job !
Watch my comments below!

File: submissions/vakulinina/memory-pair-game/main.js (hunk 1 above)
Comment (vakulinina) [reply submissions/vakulinina/memory-pair-game/main.js:22]: I used to have a bad habit of using `let` **everywhere** back then :) will fix.

**Review by babayK0 — APPROVED** [review]

**Pull request merged**